From e2efe80826b12f128e21dfffa319259a849aabac Mon Sep 17 00:00:00 2001
From: Kyle Meyer <kyle@kyleam.com>
Date: Fri, 2 Apr 2021 17:00:03 -0400
Subject: [PATCH] TST: clone: Ensure valid identity when patching HOME

test_expanduser() is failing in the Conda build for 14.1 with

    fatal: unable to auto-detect email address (got 'conda@9bc15aa8a5d2.(none)')

It's not clear why this issue has only just shown up, but
test_expanduser() can't rely on user.name and user.email being set
because it switches away from the test home set up by
datalad.setup_package().  Set up user.name and user.email in the new
home to ensure that there is a usable identity.

Re: https://github.com/conda-forge/datalad-feedstock/pull/55
---
 datalad/core/distributed/tests/test_clone.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/datalad/core/distributed/tests/test_clone.py b/datalad/core/distributed/tests/test_clone.py
index 3f1660dd03..d05fd494ba 100644
--- a/datalad/core/distributed/tests/test_clone.py
+++ b/datalad/core/distributed/tests/test_clone.py
@@ -573,6 +573,13 @@ def test_expanduser(srcpath, destpath):
     src = Dataset(Path(srcpath) / 'src').create()
     dest = Dataset(Path(destpath) / 'dest').create()
 
+    # We switch away from home set up in datalad.setup_package(), so make sure
+    # we have a valid identity.
+    with open(op.join(srcpath, ".gitconfig"), "w") as fh:
+        fh.write("[user]\n"
+                 "name = DataLad oooooTester\n"
+                 "email = test@example.com\n")
+
     with chpwd(destpath), patch.dict('os.environ', get_home_envvars(srcpath)):
         res = clone(op.join('~', 'src'), 'dest', result_xfm=None, return_type='list',
                     on_failure='ignore')
-- 
2.31.1.192.g0881477623

